### PR TITLE
[Feature] Flush and WorkManager

### PR DESCRIPTION
**Pull request by desusai7**
## Description of the change

Flush and WorkManager

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-android/109)

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -42,6 +42,9 @@ dependencies {
     testImplementation "org.powermock:powermock-core:2.0.9"
     testImplementation "org.powermock:powermock-module-junit4:2.0.9"
     testImplementation "org.powermock:powermock-api-mockito2:2.0.9"
+
+    implementation "androidx.work:work-runtime:2.7.1"
```
Comment (itsdebs) [inline core/build.gradle:46]: Make this "compileOnly"
If the developer wants this feature, they need to implement this.

Comment (desusai7) [reply core/build.gradle]: done

File: core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java [hunk 2]
```diff
@@ -40,15 +42,21 @@
...
     private String authHeaderString;
     private String anonymousIdHeaderString;
-    private int versionCode;
+    private Context context;
     private RudderConfig config;
     private DBPersistentManager dbManager;
     private RudderServerConfigManager configManager;
     private RudderPreferenceManager preferenceManager;
     private RudderEventFilteringPlugin rudderEventFilteringPlugin;
+    private RudderFlushWorkManager rudderFlushWorkManager;
```
Comment (itsdebs) [inline core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java:51]: This might crash if work manager is not implemented by the user

Comment (desusai7) [reply core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java:51]: Yeah, so to crash guard this we are already checking if the WorkManager is being present on the Classpath or not over [here](https://github.com/rudderlabs/rudder-sdk-android/blob/2464a58d22db8f81f5256d4ad858fa4a2576cad9/core/src/main/java/com/rudderstack/android/sdk/core/RudderFlushWorkManager.java#L66)

**Review by itsdebs — APPROVED** [review]

**Review by itsdebs — APPROVED** [review]

**Pull request merged**